### ISSUE TITLE
[Machine Learning] How to Implement K fold Cross-Validation in Scikit-Learn

### ISSUE THREAD

**Issue by Inyrkz** [section-engineering-education/engineering-education#5223]
### Topic Suggestion
[Machine Learning] How to Implement K fold Cross-Validation in Scikit-Learn

#### Proposed title of the article
How to Implement K fold Cross-Validation in Scikit-Learn

#### Proposed article introduction
In machine learning, the most common approach is to split a dataset into the training set and the validation set. The splitting ratio could be 70:30 or 80:20. The problem with this approach is we are not sure if a good validation accuracy means the model is really good. What if the portion of the data used as validation luckily gave us a good result? If we had used another portion of the dataset as a validation set, would our model still give us a good accuracy? These are some of the things we are not sure of. K fold cross-validation provides a better way of validating our model performance. It evaluates the model using different portions of the dataset as the validation set in various iterations.

The K fold cross-validation technique is helpful when you are training your model on a small dataset. If your dataset is large (maybe over 100,000 samples), you may not have to use K fold cross-validation. The reason is you will have enough data records in your validation set to evaluate the machine learning model. Also, using the K fold cross-validation on a large dataset will take a lot of time.

#### Key takeaways 
What you will learn:
1.	How to use K fold cross-validation to measure the performance of the model.
2.	How to use the decision tree algorithm in Sci-kit Learn.
3.	How to visualize your training accuracy, training recall, training precision and training f1-score on each fold of the dataset.
4.	How to visualize your validation accuracy, validation recall, validation precision and validation f1-score on each fold of the dataset.
5. How to reduce overfitting in a decision trees model.

#### Article quality
This article is in-depth. It will teach people how to visualize a machine learning model performance. They will be able to tell if the model is overfitting or underfitting. I realized that most of the models trained using decision trees overfit. If you can visualize the training and validation performance of the model, you’ll be able to prevent this by adjusting the hyperparameters of your decision tree model.

**Comment by ahmadmardeni1** [comment]
Sounds like a helpful topic - let's please be sure it adds value beyond what is in any official docs and/or what is covered in other blog sites. (the articles should go beyond a basic explanation - and it is always best to reference any EngEd article and build upon it). @Inyrkz 

Please be attentive to grammar/readability and make sure that you put your article through a thorough editing review prior to submitting it for final approval. (There are some great free tools that we reference in EngEd resources.)
ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.

Please be sure to double-check that it does not overlap with any existing EngEd articles, articles on other blog sites, or any incoming EngEd topic suggestions (if you haven't already) to avoid any potential article closure, please reference any relevant EngEd articles in yours. - Approved
